bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -44,21 +44,29 @@
     private static final Logger LOGGER = Loggers.getLogger("PojoCodec");
     private final ClassModel<T> classModel;
     private final CodecRegistry registry;
+    private final PropertyCodecRegistry propertyCodecRegistry;
     private final DiscriminatorLookup discriminatorLookup;
     private final ConcurrentMap<ClassModel<?>, Codec<?>> codecCache;
     private final boolean specialized;
 
-
-    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry registry, final DiscriminatorLookup discriminatorLookup) {
-        this(classModel, registry, discriminatorLookup, new ConcurrentHashMap<ClassModel<?>, Codec<?>>(), shouldSpecialize(classModel));
+    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry registry, final List<PropertyCodecProvider> propertyCodecProviders,
+                  final DiscriminatorLookup discriminatorLookup) {
+        this(classModel, registry, propertyCodecProviders, null, discriminatorLookup,
+                new ConcurrentHashMap<ClassModel<?>, Codec<?>>(), shouldSpecialize(classModel));
     }
 
-    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry registry, final DiscriminatorLookup discriminatorLookup,
+    PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry registry, final List<PropertyCodecProvider> propertyCodecProviders,
+                  final PropertyCodecRegistry propertyCodecRegistry, final DiscriminatorLookup discriminatorLookup,
```

Comment:
These two constructors are confusing as in this one a non-null propertyCodecProviders is ignored if propertyCodecRegistry is also non-null.

Instead have one constructor that takes just a List<PropertyCodecProvider> and another takes just a PropertyCodecRegistry

Reply:
Done